agent/connect/uri.go

```diff
@@ -119,6 +119,13 @@ func ParseCertURI(input *url.URL) (CertURI, error) {
 				Domain:    input.Host[idx+1:],
 			}, nil
 		}
+	} else {
+		// The SPIFFE spec states that individual workloads should have a path.
+		// We assume anything that hasn't matched by now is an external service.
```

Comment:
Options:
* We could look up all the external uri and external trust domain intentions that apply to the destination service and that would give us a way to match the possible spiffe ids. This is pretty big change.
* We could never match anything ending with the domain `.consul`. I think I didn't do this because @banks mentioned that we might want to use this feature for other non-wan-joined Consul clusters. That being said, we could change this later. I like this option.

The other thing to note (and paul addresses it [here](https://github.com/hashicorp/consul/pull/6241#issuecomment-516820752)) is that when we get the certificate chain from Envoy we'll know if this was an external request vs. internal based on the signing root.